File: YoupiBanane/hello.py
#!/usr/bin/env python3
import os

method = os.environ.get("REQUEST_METHOD", "GET")
query  = os.environ.get("QUERY_STRING", "")

print("Content-Type: text/html\r\n\r\n")
print("<html><body>")
print("<h1>Hello from CGI!</h1>")
print("<p>Method: " + method + "</p>")
print("<p>Query: " + query + "</p>")
print("</body></html>")

File: YoupiBanane/hello.php
#!/usr/bin/env php-cgi
<?php
$method = getenv("REQUEST_METHOD") ?: "GET";
$query  = getenv("QUERY_STRING") ?: "";

// 读取 POST body
$body = "";
if ($method === "POST") {
    $body = file_get_contents("php://stdin");
}

echo "Content-Type: text/html\r\n\r\n";
?>
<html>
<body>
  <h1>Hello from PHP CGI!</h1>
  <p>Method: <?php echo htmlspecialchars($method); ?></p>
  <p>Query: <?php echo htmlspecialchars($query); ?></p>
  <?php if ($body !== ""): ?>
  <p>Body: <?php echo htmlspecialchars($body); ?></p>
  <?php endif; ?>
</body>
</html>
